usr/share/rear/format/USB/default/300_format_usb_disk.sh

```diff
@@ -53,56 +51,29 @@ if ! test $boot_partition_flag ; then
-    local efi_partition_size_bytes=$(( USB_UEFI_PART_SIZE * MiB_bytes ))
-    # The end byte is the last byte that belongs to that partition so that one must be careful to use "start_byte + partition_size_in_bytes - 1":
-    local efi_partition_end_byte=$(( efi_partition_start_byte + efi_partition_size_bytes - 1 ))
-    if ! parted -s $RAW_USB_DEVICE unit B mkpart primary $efi_partition_start_byte $efi_partition_end_byte ; then
-        Error "Failed to create EFI system partition $RAW_USB_DEVICE$current_partition_number"
-    fi
-    # Partition 1 is the EFI system partition (vfat partition) on which EFI/BOOT/BOOTX86.EFI resides
-    # so the number of the partition that can be set up next has to be one more (i.e. now 2):
-    current_partition_number=$(( current_partition_number + 1 ))
-    # Calculate byte value for the start of the subsequent data partition:
-    data_partition_start_byte=$(( efi_partition_end_byte + 1 ))
-
-    # End of EFI case.
-else
-    # Begin non-EFI case:
-
-    # If not set use fallback value 'msdos' (same as the default value in default.conf):
-    test "msdos" = "$USB_DEVICE_PARTED_LABEL" -o "gpt" = "$USB_DEVICE_PARTED_LABEL" || USB_DEVICE_PARTED_LABEL="msdos"
-    LogPrint "Creating partition table of type $USB_DEVICE_PARTED_LABEL on $RAW_USB_DEVICE"
-    if ! parted -s $RAW_USB_DEVICE mklabel $USB_DEVICE_PARTED_LABEL ; then
-        Error "Failed to create $USB_DEVICE_PARTED_LABEL partition table on $RAW_USB_DEVICE"
-    fi
+### Create partition table section
 
-    # USB_PARTITION_ALIGN_BLOCK_SIZE is the first byte of the boot partition:
-    local boot_partition_start_byte=$(( USB_PARTITION_ALIGN_BLOCK_SIZE * MiB_bytes ))
+# Initialize USB disk via "parted mklabel" (create partition table)
+# If not set use fallback value 'msdos' (same as the default value in default.conf):
+test "msdos" = "$USB_DEVICE_PARTED_LABEL" -o "gpt" = "$USB_DEVICE_PARTED_LABEL" || USB_DEVICE_PARTED_LABEL="msdos"
+LogPrint "Creating partition table of type $USB_DEVICE_PARTED_LABEL on $RAW_USB_DEVICE"
+if ! parted -s $RAW_USB_DEVICE mklabel $USB_DEVICE_PARTED_LABEL ; then
+    Error "Failed to create $USB_DEVICE_PARTED_LABEL partition table on $RAW_USB_DEVICE"
+fi
 
+### Create partitions section
+# in order:
+# * BIOS partition (GPT only) / partition gap aka empty space for MSDOS
+# * EFI (in case of EFI)
+# * boot partition (optional but encuraged)
+# * backup/storage/data partition
+
+# In case of GPT with BIOS boot we need a 'bios_grub' partition
+# In case of MSDOS partition table we just need some free space in between the partition table and the first partition
+# This partition should be the first one afaik this has a better chance to work with odd BIOS firmwares
+if is_true "$FORMAT_BIOS" ; then
     if [[ "$USB_DEVICE_PARTED_LABEL" == "gpt" ]] ; then
+        LogPrint "The --bios toggle was used with format - making an BIOS bootable device $RAW_USB_DEVICE"
```

Comment:
You may leave out "The --bios toggle was used with format - " and only show
"Making a BIOS bootable device $RAW_USB_DEVICE"
(no 'an' in case of BIOS)
because plain rear format /dev/sdX (i.e. without any --bios toggle)
also makes it BIOS bootable.